c7n/policy.py

```diff
@@ -73,8 +75,13 @@ def __init__(self, data, options):
         # We store all the policies passed in so we can refilter later
         self._all_policies = []
         for p in self.data.get('policies', []):
-            self._all_policies.append(
-                Policy(p, options, session_factory=self.test_session_factory()))
+            if 'all' in options.regions:
+                options.regions = utils.get_available_regions(p['resource'])
+            for region in options.regions:
+                options_copy = copy.copy(options)
+                options_copy.region = region
+                self._all_policies.append(
```

Comment:
@kapilt  - I fixed this - invalid region/resource combos are now ignored.  Do you want a warning message?  I don't have one currently, but I suspect we should because this scenario will only happen when the user explicitly asked for this combo.